owrx/websocket.py

```diff
@@ -175,60 +177,67 @@ def read_loop(self):
-            if self.handler.rfile in read:
-                available = True
+            try:
+                header = protected_read(2)
                 self.resetPing()
-                while self.open and available:
+                opcode = header[0] & 0x0F
+                length = header[1] & 0x7F
+                mask = (header[1] & 0x80) >> 7
+                if length == 126:
+                    header = protected_read(2)
+                    length = (header[0] << 8) + header[1]
+                if mask:
+                    masking_key = protected_read(4)
+                data = None
+                if length > 0:
+                    data = protected_read(length)
+                    if mask:
+                        data = bytes([b ^ masking_key[index % 4] for (index, b) in enumerate(data)])
+                if opcode == OPCODE_TEXT_MESSAGE:
+                    message = data.decode("utf-8")
                     try:
-                        header = protected_read(2)
-                        opcode = header[0] & 0x0F
-                        length = header[1] & 0x7F
-                        mask = (header[1] & 0x80) >> 7
-                        if length == 126:
-                            header = protected_read(2)
-                            length = (header[0] << 8) + header[1]
-                        if mask:
-                            masking_key = protected_read(4)
-                        data = protected_read(length)
-                        if mask:
-                            data = bytes([b ^ masking_key[index % 4] for (index, b) in enumerate(data)])
-                        if opcode == OPCODE_TEXT_MESSAGE:
-                            message = data.decode("utf-8")
-                            try:
-                                self.messageHandler.handleTextMessage(self, message)
-                            except Exception:
-                                logger.exception("Exception in websocket handler handleTextMessage()")
-                        elif opcode == OPCODE_BINARY_MESSAGE:
-                            try:
-                                self.messageHandler.handleBinaryMessage(self, data)
-                            except Exception:
-                                logger.exception("Exception in websocket handler handleBinaryMessage()")
-                        elif opcode == OPCODE_PING:
-                            self.sendPong()
-                        elif opcode == OPCODE_PONG:
-                            # since every read resets the ping timer, there's nothing to do here.
-                            pass
-                        elif opcode == OPCODE_CLOSE:
-                            logger.debug("websocket close frame received; closing connection")
-                            self.open = False
-                        else:
-                            logger.warning("unsupported opcode: {0}".format(opcode))
-                    except Drained:
```

Comment:
`Drained` exception is no longer used, so the class definition (further up) should be removed as well